main.ts

```diff
@@ -1,339 +1,377 @@
-    try {
-        win.webContents.send(channel, ...args);
-    }
-    catch (error) {
-        console.error(`could not send to '${channel}' with args '${JSON.stringify(args)}`);
-    }
+function send(channel: string, ...additionalArgs: any[]): void {
+  try {
+    win.webContents.send(channel, ...additionalArgs)
+  } catch (error) {
+    console.error(`could not send to '${channel}' with args '${JSON.stringify(args)}`)
+  }
 }
 
-launch.register(ipcMain);
+launch.register(ipcMain)
 
 update.register(ipcMain, (event, autoDownload) => {
-    switch (event) {
-        case update.AutoUpdaterEvent.UpdateAvailable:
-            tray?.displayBalloon({
-                iconType: 'info',
-                title: 'New update available',
-                content: 'A new update is available. Will be automatically downloaded unless otherwise specified.',
-            });
-            if (!autoDownload && !downloadItem) {
-                downloadItem = new MenuItem({
-                    label: 'Download Update',
-                    type: 'normal',
-                    click: () => {
-                        update.download();
-                        downloadItem.enabled = false;
-                    }
-                });
-                menu?.insert(2, downloadItem);
-            }
-            break;
-        case update.AutoUpdaterEvent.UpdateDownloaded:
-            tray?.displayBalloon({
-                iconType: 'info',
-                title: 'Update ready to install',
-                content: 'The new update is now ready to install. Please relaunch your application.',
-            });
-            break;
+  switch (event) {
+    case update.AutoUpdaterEvent.UpdateAvailable:
+      tray?.displayBalloon({
+        iconType: 'info',
+        title: 'New update available',
+        content:
+          'A new update is available. Will be automatically downloaded unless otherwise specified.',
+      })
+      if (!autoDownload && !downloadItem) {
+        downloadItem = new MenuItem({
+          label: 'Download Update',
+          type: 'normal',
+          click: () => {
+            update.download()
+            downloadItem.enabled = false
+          },
+        })
+        menu?.insert(2, downloadItem)
+      }
+      break
+    case update.AutoUpdaterEvent.UpdateDownloaded:
+      tray?.displayBalloon({
+        iconType: 'info',
+        title: 'Update ready to install',
+        content: 'The new update is now ready to install. Please relaunch your application.',
+      })
+      break
+  }
+  send(event)
+})
+
+robot.register(ipcMain)
+
+game.register(ipcMain, (poe) => {
+  send('game-active-change', serve ? true : poe.active)
+  // send('game-active-change', poe.active);
+
+  if (win) {
+    if (poe.active) {
+      if (process.platform !== 'linux') {
+        win.setAlwaysOnTop(true, 'pop-up-menu', 1)
+        win.setVisibleOnAllWorkspaces(true)
+      } else {
+        win.setAlwaysOnTop(true)
+        win.maximize()
+      }
```

Comment:
this is the hack that loads the window, apparently windows do not get maximized with setAlwaysOnTop only so I had to do that manually

Reply:
Making sure I understand what `win.maximize()` does - if someone is playing in windowed mode, but not fullscreen, will this resize their game window?

Reply:
```win.maximize()``` just maximizes the poe overlay clickthrough window on which the remaining application is drawn on, we don't have access to the ```level``` parameter on linux and because of that the window would start minimized. Since that's in conflict with the remainder of the overlay windows, I had to do that manually. This won't interact with the game in any way. The ```level``` parameter does something similar on windows, it pulls the overlay window into forground and thus it doesn't get minimized.

Reply:
Right, thanks for the explanation.